client/components/tinymce/plugins/wpcom-view/plugin.js

```diff
@@ -854,23 +856,32 @@ function wpview( editor ) {
 
 	editor.addButton( 'wp_view_remove', {
 		tooltip: i18n.translate( 'Remove' ),
-		icon: 'dashicon dashicons-no',
+		icon: 'dashicon dashicons-trash',
 		onClick: function() {
 			selected && removeView( selected );
 		}
 	} );
 
 	editor.once( 'preinit', function() {
-		toolbar = editor.wp._createToolbar( [
+		editToolbar = editor.wp._createToolbar( [
 			'wp_view_edit',
 			'wp_view_remove'
 		] );
+		removeToolbar = editor.wp._createToolbar( [
+			'wp_view_remove'
+		] );
 	} );
 
 	editor.on( 'wptoolbar', function( event ) {
 		if ( selected ) {
+
+			// Checks for ids in the gallery shortcode
+			const { attrs: { named: { ids } } } = Shortcode.parse(
+				decodeURIComponent( editor.dom.getAttrib( selected, 'data-wpview-text' ) )
+			);
+
 			event.element = selected;
-			event.toolbar = toolbar;
+			event.toolbar = ids ? editToolbar : removeToolbar;
```

Comment:
If I'm not missing something, this change means that all `wpview` that aren't shortcodes, and all that are but don't have `ids` attribute will get the `removeToolbar`. Don't think this was intended?